conda_lock/conda_lock.py

```diff
@@ -23,42 +31,130 @@
 DEFAULT_PLATFORMS = ["osx-64", "linux-64", "win-64"]
 
 
-def solve_specs_for_arch(channels: List[str], specs: List[str], platform: str) -> str:
+def ensure_conda(conda_executable: Optional[str]):
+    if conda_executable:
+        if pathlib.Path(conda_executable).exists():
+            return conda_executable
+
+    conda_executable = shutil.which("conda")
+    if conda_executable:
+        return conda_executable
+
+    conda_executable = shutil.which("conda.exe")
+    if conda_executable:
+        return conda_executable
+
+    logging.info(
+        "No existing conda installation found.  Installing the standalone conda solver"
+    )
+    return install_conda_exe()
+
+
+def install_conda_exe():
+    conda_exe_prefix = "https://repo.anaconda.com/pkgs/misc/conda-execs"
```

Comment:
Are those advertised and/or recommended for use? That is a great idea but I'm worried if they are well maintained.

Reply:
Talked with jjhelmus and they are released and maintained in the same cadence as Anaconda distribution, so it should be fine

Reply:
That is great news!